src/collection.js

```diff
@@ -889,11 +891,17 @@ export default class Collection {
 
         // The result of a batch returns data and permissions.
         // XXX: permissions are ignored currently.
-        const conflicts = synced.conflicts.map(({type, local, remote}) => {
+        return Promise.all(synced.conflicts.map(({type, local, remote}) => {
           // Note: we ensure that local data are actually available, as they may
           // be missing in the case of a published deletion.
-          return {type, local: local && local.data || {}, remote};
-        });
+          const safeLocal = local && local.data || {};
+          return this._decodeRecord("remote", safeLocal).then(realLocal => {
+            return this._decodeRecord("remote", remote).then(realRemote => {
+              return {type, local: realLocal, remote: realRemote};
+            });
+          });
```

Comment:
This may improve legibility a little:

```js
const safeLocal = local && local.data || {};
return this._decodeRecord("remote", safeLocal)
  .then(realLocal => {
    return this._decodeRecord("remote", remote);
  })
  .then(realRemote => {
    return {type, local: realLocal, remote: realRemote};
  });
```